googleapiclient/http.py

```diff
@@ -91,12 +95,24 @@ def _retry_request(http, num_retries, req_type, sleep, rand, uri, method, *args,
 
     try:
       resp, content = http.request(uri, method, *args, **kwargs)
-    except ssl.SSLError:
+    except (ssl.SSLError, TimeoutError) as ex:
       if retry_num == num_retries:
         raise
       else:
+        logging.warning('Encountered exception %s', repr(ex))
         continue
-    if resp.status < 500:
+    reason = ''
+    try:
+      if resp and resp.status == 403:
+        data = json.loads(content.decode('utf-8'))
+        reason = data['error']['errors'][0]['reason']
+        logging.warning('Encountered status %d with reason "%s"' %
+                        (resp.status, reason))
+    except Exception:
```

Comment:
iirc `json.loads` only throws `ValueError`. In any case, you should scope the `try: except` as tightly as possible:

```python
if resp and resp.status == 403:
    try:
        data = json.loads(content.decode('utf-8'))
        reason = data['error']['errors'][0]['reason']
    except ValueError, KeyError:
        logging.warning('Invalid JSON content from response')
        reason = content.decode('utf-8')

    logging.warning('Encountered status %d with reason "%s"' %
        (resp.status, reason))
```